typescript/cli/src/deploy/warp.ts

```diff
@@ -156,69 +158,54 @@ async function runBuildConfigStep({
-    },
-  };
-
-  for (const synthetic of synthetics) {
-    const sChainName = synthetic.chainName;
-    configMap[sChainName] = {
-      type: TokenType.synthetic,
-      name: synthetic.name || baseMetadata.name,
-      symbol: synthetic.symbol || baseMetadata.symbol,
-      totalSupply: synthetic.totalSupply || 0,
-      owner,
-      mailbox: synthetic.mailbox || mergedContractAddrs[sChainName].mailbox,
-      interchainSecurityModule:
-        synthetic.interchainSecurityModule ||
-        mergedContractAddrs[sChainName]?.interchainSecurityModule ||
-        mergedContractAddrs[sChainName]?.multisigIsm,
-      // ismFactory: mergedContractAddrs[sChainName].domainRoutingIsmFactory, // TODO fix
-      foreignDeployment: synthetic.foreignDeployment,
-    };
-  }
-
-  // Request input for any address fields that are missing
   const requiredRouterFields: Array<keyof ConnectionClientConfig> = ['mailbox'];
-  let hasShownInfo = false;
-  for (const [chain, token] of Object.entries(configMap)) {
+  const remotes: string[] = [];
+
+  /// @dev This will keep track of the base collateral metadata which can get overwritten if there are multiple collaterals.
+  /// These 'base' variables are used to derive synthetic fields
+  /// @todo Remove this artifact when multi-collateral is enabled
+  let baseChainName = '';
+  let baseMetadata = {} as MinimalTokenMetadata;
+  // Create config that coalesce together values from the config file,
+  for (const [chain, config] of Object.entries(warpRouteConfig)) {
+    const mergedContractAddrs = getMergedContractAddresses(
+      coreArtifacts,
+      Object.keys(warpRouteConfig),
+    );
+    // the artifacts, and the SDK as a fallback
+    config.owner = owner;
+    config.mailbox = config.mailbox || mergedContractAddrs[chain]?.mailbox;
+    config.interchainSecurityModule =
+      config.interchainSecurityModule ||
+      mergedContractAddrs[chain]?.interchainSecurityModule ||
+      mergedContractAddrs[chain]?.multisigIsm;
+    // config.ismFactory: mergedContractAddrs[baseChainName].domainRoutingIsmFactory, // TODO fix when updating from routingIsm
+
+    if (isCollateralConfig(config) || isNativeConfig(config)) {
+      // Store the base metadata
+      baseChainName = chain;
+      baseMetadata = await fetchBaseTokenMetadata(chain, config, multiProvider);
+      log(
+        `Using token metadata: Name: ${baseMetadata.name}, Symbol: ${baseMetadata.symbol}, Decimals: ${baseMetadata.decimals}`,
+      );
+      if (isCollateralConfig(config)) {
+        config.name = baseMetadata.name;
+        config.symbol = baseMetadata.symbol;
+        config.decimals = baseMetadata.decimals;
+      }
+    } else if (isSyntheticConfig(config)) {
+      // Use the config, or baseMetadata
+      config.name = config.name || baseMetadata.name;
+      config.symbol = config.symbol || baseMetadata.symbol;
+      config.totalSupply = config.totalSupply || 0;
+      remotes.push(chain);
+    }
```

Comment:
this is duped with the token deployer logic right?

Reply:
I dont see how this is avoiding the overwriting tbh

Reply:
ditto